lib/HLSL/DxilValidation.cpp

```diff
@@ -3988,6 +3988,114 @@ static void ValidateMetadata(ValidationContext &ValCtx) {
   }
 
   const hlsl::ShaderModel *SM = ValCtx.DxilMod.GetShaderModel();
+  // validate that any wavesize tags don't appear outside their expected shader
+  // models. Validate only 1 tag exists per entry point.
+  // Don't do this validation if the shader is non-compute
+  if (SM->IsCS() || SM->IsLib()) {
+
+    NamedMDNode *TA = pModule->getNamedMetadata("dx.entryPoints");
```

Comment:
I'm curious why the variable is `TA`? Was it just copied from another loaded MD node?

Reply:
I copied / was inspired to write the code the way it was by ValidateTypeAnnotation(). I suppose TA stands for TypeAnnotation. I'll perform some renames, because this has nothing to do with the type annotations :)